core/util.py

```diff
@@ -157,3 +253,23 @@ def custom_serialiser(obj: Any) -> str:
         return obj.isoformat()
     else:
         raise TypeError(f"Cannot serialise object of type {type(obj)}")
+
+
+def move_event_data_to_json_blob(
```

Comment:
this could have a more generic name? It doesn't need to be specific to "event data" since it actually just moves any dataframe columns into a dict? And something outside this func dumps into the JSON? Probs need to update the signature and the docstring to reflect this, and maybe explain the structure of the json/dict (ie columns/values moved to be key/value pairs in the new dict column)?